modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -236,11 +236,13 @@ private void rolloverDataStream(RolloverRequest rolloverRequest, ActionListener<
         client.admin().indices().rolloverIndex(rolloverRequest, new ActionListener<>() {
             @Override
             public void onResponse(RolloverResponse rolloverResponse) {
-                logger.info(
-                    "DLM successfully rolled over datastream [{}]. The new index is [{}]",
-                    rolloverTarget,
-                    rolloverResponse.getNewIndex()
-                );
+                if (rolloverResponse.isRolledOver()) {
```

Comment:
I was thinking to add a small comment about this, because I thought that a successful response would mean that index got rolled over, but of course on a second thought, that's not true :).
```suggestion
                // Log only when the conditions were met and the index was rolled over.
                if (rolloverResponse.isRolledOver()) {
```